tensorboard/webapp/plugins/npmi/util/sort_annotations.ts

```diff
@@ -15,47 +15,126 @@ limitations under the License.
+      for (const annotation of keys) {
+        minData[annotation] = Math.min(
+          ...annotationData[annotation]
+            .filter((annotation) => annotation.metric === strippedMetric)
+            .map((filtered) =>
+              filtered.nPMIValue === null ? Infinity : filtered.nPMIValue
+            )
+        );
+      }
+      keys = keys.sort((a, b) => {
+        return minData[a] - minData[b];
+      });
+      break;
+    }
+  }
+  return keys;
+}
+
+function embeddingSort(
+  embeddingData: EmbeddingListing,
+  keys: string[],
+  sort: AnnotationSort
+): string[] {
+  switch (sort.order) {
+    case SortOrder.SIMILAR: {
+      const distance: {[annotation: string]: number} = {};
+      for (const annotation of keys) {
+        if (annotation === sort.metric) {
+          distance[annotation] = Number.NEGATIVE_INFINITY;
+        } else {
+          distance[annotation] = embeddingData[annotation]
+            ? calculateDistance(
+                embeddingData[sort.metric],
+                embeddingData[annotation]
+              )
+            : Number.POSITIVE_INFINITY;
+        }
+      }
+      keys = keys.sort((a, b) => {
+        return distance[a] - distance[b];
+      });
+      break;
     }
-    result = result.sort((a, b) => {
-      return maxData[b] - maxData[a];
-    });
-  } else if (sort.order === SortOrder.UP) {
-    const minData: {[annotation: string]: number} = {};
-    for (const annotation of result) {
-      minData[annotation] = Math.min(
-        ...annotationData[annotation]
-          .filter((annotation) => annotation.metric === strippedMetric)
-          .map((filtered) =>
-            filtered.nPMIValue === null ? Infinity : filtered.nPMIValue
-          )
-      );
+    case SortOrder.DISSIMILAR: {
+      const distance: {[annotation: string]: number} = {};
+      for (const annotation of keys) {
+        if (annotation === sort.metric) {
+          distance[annotation] = Number.POSITIVE_INFINITY;
+        } else {
+          distance[annotation] = embeddingData[annotation]
+            ? calculateDistance(
+                embeddingData[sort.metric],
+                embeddingData[annotation],
+                Number.NEGATIVE_INFINITY
+              )
+            : Number.NEGATIVE_INFINITY;
+        }
+      }
+      keys = keys.sort((a, b) => {
+        return distance[b] - distance[a];
+      });
+      break;
     }
-    result = result.sort((a, b) => {
-      return minData[a] - minData[b];
-    });
   }
-  return result;
+  return keys;
+}
+
+function calculateDistance(
+  reference: number[],
+  toEmbedding: number[],
+  faultCase: number = Number.POSITIVE_INFINITY
+): number {
+  if (reference.length != toEmbedding.length) return faultCase;
+  const subtracted = toEmbedding.map((i, n) => i - reference[n]);
+  const powered = subtracted.map((e) => Math.pow(e, 2));
+  const sum = powered.reduce((total, current) => total + current, 0);
+  return Math.sqrt(sum);
```

Comment:
I forget but is sqrt important for the cartesian distance? If it is only used for sorting, do you need to sqrt?

Reply:
In our case, we can remove it, as we dont use the exact distance for anything.

Reply:
It only transforms the domain that the function operates on, so not needed in our case.